core/src/main/scala/org/ensime/core/StructureViewBuilder.scala

```diff
@@ -0,0 +1,84 @@
+package org.ensime.core
+
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable.ListBuffer
+import scala.reflect.io.AbstractFile
+import scala.tools.refactoring.common.{ CompilerAccess, PimpedTrees }
+import scala.reflect.internal.util.SourceFile
+import org.ensime.api._
+
+class StructureViewBuilder(val global: RichPresentationCompiler)
```

Comment:
the pattern we've been following to date has been to mix in, rather than take as a constructor parameter. Can you please require a mixin from `RichPresentationCompiler`?